### PR TITLE
faq/tidb-faq: update out of date faqs

### LINKED ISSUE #2325: Out of date FAQs
## Error Report

These  FAQs are out of date, and not part of the challenge program, so I am filing a bug report here.

1. What is the URL/path of the document related to this issue?

https://pingcap.com/docs/stable/faq/tidb/

2. How would you like to improve it?

> TiDB show processlist displays the estimated memory usage (unit: Byte) of the current session, which is not displayed in MySQL show processlist.

SHOW PROCESSLIST no longer shows memory (not mysql compatible behavior). The memory stats are in information_schema.processlist though.

> What's the maximum number of concurrent connections that TiDB supports?
> The current TiDB version has no limit for the maximum number of concurrent connections. If too large concurrency leads to an increase of response time, you can increase the capacity by adding TiDB nodes.

This is now config dependent - max_connections. Default is still unlimited.

> Does TiDB support CBO (Cost-Based Optimization)? If yes, to what extent?
> Yes. TiDB uses the cost-based optimizer. The cost model and statistics are constantly optimized. TiDB also supports correlation algorithms like hash join and soft merge.

It should be sort merge, not soft merge. I believe "correlation" was meant to be join as well.

### PR DESCRIPTION
**Pull request by nullnotnil**
### What is changed, added or deleted? (Required)

Updates out of date FAQs

Fixes #2325

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

Fixes #2325

### CODE REVIEW

File: faq/tidb-faq.md [hunk 1]
```diff
@@ -1051,7 +1051,7 @@ From the above results, you can get that the `add index` operation is being proc
 
 #### Does TiDB support CBO (Cost-Based Optimization)? If yes, to what extent?
 
-Yes. TiDB uses the cost-based optimizer. The cost model and statistics are constantly optimized. TiDB also supports correlation algorithms like hash join and soft merge.
+Yes. TiDB uses the cost-based optimizer. The cost model and statistics are constantly optimized. TiDB also supports join algorithms like hash join and sort merge.
```
Comment (zz-jason) [inline faq/tidb-faq.md:1054]: how about:

```suggestion
Yes. TiDB uses the cost-based optimizer. The cost model and statistics are constantly optimized. TiDB also supports join algorithms like hash join and sort-merge join.
```

File: faq/tidb-faq.md [hunk 2]
```diff
@@ -539,7 +539,7 @@ In the communication process between the TiDB server and the TiKV server, the `S
 
 #### What's the maximum number of concurrent connections that TiDB supports?
 
-The current TiDB version has no limit for the maximum number of concurrent connections. If too large concurrency leads to an increase of response time, you can increase the capacity by adding TiDB nodes.
+By default, there is no limit on the maximum number of connections per TiDB server. A limit may be enforced by setting `max-server-connections` in the config.toml file. If too large concurrency leads to an increase of response time, it is recommended to increase the capacity by adding TiDB nodes.
```
Comment (ran-huang) [inline faq/tidb-faq.md:542]:
```suggestion
By default, there is no limit on the maximum number of connections per TiDB server. A limit may be enforced by setting `max-server-connections` in the `config.toml` file. If too large concurrency leads to an increase of response time, it is recommended to increase the capacity by adding TiDB nodes.
```

**Pull request merged**